cordformation/src/main/kotlin/net/corda/plugins/External.kt

```diff
@@ -0,0 +1,28 @@
+package net.corda.plugins;
+
+import org.gradle.api.tasks.Input
+
+open class External {
+
+    @get:Input
+    var containerName: String = "external-service"
+
+    @get:Input
+    var servicePorts: List<Int> = listOf(8080)
+
+    @get:Input
+    var containerImage = ""
+
+    @get:Input
+    var volumes: List<Map<String, String>> = emptyList()
+
+    @get:Input
+    var environment: Map<String, Any> = emptyMap()
```

Comment:
This should be a `MapProperty`, which is also automatically empty by default and so doesn't need to be `@Optional`.